docs/control-structures.rst

```diff
@@ -625,21 +627,23 @@ The error code supplied with the error data indicates the kind of panic.
 #. 0x41: If you allocate too much memory or create an array that is too large.
 #. 0x51: If you call a zero-initialized variable of internal function type.
 
-The ``require`` function either creates an error without any data or
-an error of type ``Error(string)``. It
-should be used to ensure valid conditions
-that cannot be detected until execution time.
-This includes conditions on inputs
-or return values from calls to external contracts.
+The ``require`` function provides three overloads:
+
+1. ``require(bool)`` which only accepts a boolean condition.
+2. ``require(bool, string)`` which accepts a boolean condition, as well as an error message.
+3. ``require(bool, error)`` which accepts a boolean condition, and a custom error.
+
+The first overload (condition only), will create an error without any data.
+The second overload will create an ``Error(string)``, whereas the third overload will create
+a custom, user supplied error, provided as the second argument to the ``require`` function.
```

Comment:
Why not have the list immediately say for each case what error signature is created rather than which arguments are accepted, which seems a bit redundant?